### PR TITLE
Add support for volumes to mount host paths

### PR DESCRIPTION
**Pull request by qaifshaikh**
Addresses https://github.com/hercules-ci/arion/issues/188

### CODE REVIEW

**Comment by qaifshaikh** [comment]
We use this already for colima in our setups. It works quite well, but we've patched it to an old revision of Marion
Precisely https://github.com/hercules-ci/arion/blob/a95066a8449765df2ad9c3a0d0c8307f5f52cbd7/src/nix/modules/composition/docker-compose.nix

so I'll need someone to share their input on this please 🙏🏻

File: src/nix/modules/composition/docker-compose.nix [hunk 1]
```diff
@@ -63,6 +63,11 @@ in
       type = lib.types.attrsOf (lib.types.submodule service);
       description = "An attribute set of service configurations. A service specifies how to run an image as a container.";
     };
+    docker-compose.volumes = lib.mkOption {
+      type = lib.types.anything;
```
Comment (roberth) [inline src/nix/modules/composition/docker-compose.nix:67]: The merge function of `anything` can result in surprising behavior. I'd find a submodule containing the bare minimum of options to be a more maintainable solution, as I won't have to guess what behaviors users actually rely on, when reviewing improvements later.

Comment (qaifshaikh) [reply src/nix/modules/composition/docker-compose.nix:67]: Sounds good!

Comment (qaifshaikh) [reply src/nix/modules/composition/docker-compose.nix:67]: Should i go with what is currently used for services?

https://github.com/KeepTruckin/arion/blob/issue-188/volumes-support/src/nix/modules/service/docker-compose-service.nix#L56-L60

File: src/nix/modules/composition/docker-compose.nix [hunk 2]
```diff
@@ -63,6 +63,11 @@ in
       type = lib.types.attrsOf (lib.types.submodule service);
       description = "An attribute set of service configurations. A service specifies how to run an image as a container.";
     };
+    docker-compose.volumes = lib.mkOption {
+      type = lib.types.attrsOf (lib.types.submodule service.volumes);
```
Comment (qaifshaikh) [inline src/nix/modules/composition/docker-compose.nix:67]: @roberth does this work? 🤔

**Comment by qaifshaikh** [comment]
nudge @roberth for review

**Comment by srid** [comment]
This would be good to have. I've been using `docker-compose.raw.volumes`.

**Review by roberth — APPROVED** [review]
Documentation and merging for volumes would be nice, but this is good as a first step.

**Pull request merged**